src/Storages/StorageMergeTreeIndex.cpp

```diff
@@ -233,18 +254,33 @@ StorageMergeTreeIndex::StorageMergeTreeIndex(
     const StorageID & table_id_,
     const StoragePtr & source_table_,
     const ColumnsDescription & columns,
-    bool with_marks_)
+    bool with_marks_,
+    bool with_minmax_)
     : IStorage(table_id_)
     , source_table(source_table_)
     , with_marks(with_marks_)
+    , with_minmax(with_minmax_)
 {
     const auto * merge_tree = dynamic_cast<const MergeTreeData *>(source_table.get());
     if (!merge_tree)
         throw Exception(ErrorCodes::BAD_ARGUMENTS, "Storage MergeTreeIndex expected MergeTree table, got: {}", source_table->getName());
 
     data_parts = merge_tree->getDataPartsVectorForInternalUsage();
+    std::erase_if(data_parts, [](const MergeTreeData::DataPartPtr & part) { return part->isEmpty(); });
```

Comment:
Maybe a stupid question: why is this `erase_if` needed? Is it an optimization?

Reply:
This serves as a safety mechanism ensuring we access fields that are guaranteed to exist in non-empty parts.